### PR TITLE
Fixes Typing errors for points at infinity (NoneTypes)

### PR DESCRIPTION
**Pull request by CarlBeek**
### What was wrong?

The point at infinity on any given curve is encoded as `None`. MyPy was not informed.

### How was it fixed?

I told MyPy.

#### Cute Animal Picture

image

### CODE REVIEW

File: py_ecc/bls12_381/bls12_381_curve.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def is_inf(pt: GeneralPoint[Field]) -> bool:
 def is_on_curve(pt: Point2D[Field], b: Field) -> bool:
     if is_inf(pt):
         return True
-    x, y = pt
+    x, y = pt  # type: ignore[misc]
```
Comment (CarlBeek) [inline py_ecc/bls12_381/bls12_381_curve.py]: If anyone knows why mypy is so unhappy here, please let me know. The `if is_inf(pt)` check above checks for `None` so there is no way to reach line 66 with `pt = None`. MyPy doesn't realise this for some reason. :/

File: py_ecc/fields/field_elements.py [hunk 2]
```diff
@@ -201,7 +201,7 @@ class FQP(object):
 
     def __init__(self,
                  coeffs: Sequence[IntOrFQ],
-                 modulus_coeffs: Sequence[IntOrFQ] = None) -> None:
+                 modulus_coeffs: Sequence[IntOrFQ] = []) -> None:
```
Comment (CarlBeek) [inline py_ecc/fields/field_elements.py]: This is to handle a separate error where `len(modulus_coeffs)` is called in line 208 below. This is not possible with `None` as it has no `__len__`. The reason for having a default argument here is for simple `type(self)` calls.

File: py_ecc/typing.py [hunk 3]
```diff
@@ -64,8 +65,8 @@
     bls12_381_FQ2,
     bls12_381_FQ12,
 )
-Point2D = Tuple[Field, Field]
-Point3D = Tuple[Field, Field, Field]
+Point2D = Optional[Tuple[Field, Field]]  # Point at infinity is encoded as a None
```
Comment (CarlBeek) [inline py_ecc/typing.py]: This is the meat of this PR and addresses 48 type errors.

Side note, should type checking be a part of CI on this repo? I feel like it should.

Comment (hwwhww) [reply py_ecc/typing.py:68]: I opened issue #95 for this.

**Pull request merged**